diff --git a/clients/roscpp/src/libros/io.cpp b/clients/roscpp/src/libros/io.cpp
index 004ed99ac..2e91d6ef0 100644
--- a/src/libros/io.cpp
+++ b/src/libros/io.cpp
@@ -416,9 +416,9 @@ int is_async_connected(socket_fd_t &socket, int &err) {
                (!FD_ISSET(socket, &wfds) && FD_ISSET(socket, &exceptfds)));
     if (FD_ISSET(socket, &exceptfds)) {
       // an error occurred during connection
-      int errinfo;
+      int errinfo = 0;
       socklen_t errlen = sizeof(int);
-      if (getsockopt(socket, SOL_SOCKET, SO_ERROR, &errinfo, &errlen) == -1)
+      if (getsockopt(socket, SOL_SOCKET, SO_ERROR, reinterpret_cast<char*>(&errinfo), &errlen) == -1)
       {
         // getsockopt() error
         ROSCPP_CONN_LOG_DEBUG("getsockopt() on socket[%d] failed with error [%s]",
