diff --git a/src/rviz/CMakeLists.txt b/src/rviz/CMakeLists.txt
index d5941099..d8229479 100644
--- a/src/rviz/CMakeLists.txt
+++ b/src/rviz/CMakeLists.txt
@@ -1,6 +1,7 @@
 find_package(yaml-cpp REQUIRED)
 
 include_directories(${YAML_CPP_INCLUDE_DIR})
+add_compile_definitions(YAML_CPP_DLL)
 
 if(UNIX AND NOT APPLE)
   find_package(X11 REQUIRED)

diff --git a/src/rviz/default_plugin/marker_utils.cpp b/src/rviz/default_plugin/marker_utils.cpp
index 541d2ceb..ff234436 100644
--- a/src/rviz/default_plugin/marker_utils.cpp
+++ b/src/rviz/default_plugin/marker_utils.cpp
@@ -45,6 +45,10 @@
 #include <rviz/validate_quaternions.h>
 #include <rviz/validate_floats.h>
 
+#if defined(_WIN32) && defined(DELETE)
+  #undef DELETE
+#endif
+
 namespace rviz
 {
 MarkerBase*

diff --git a/src/rviz/default_plugin/marker_display.cpp b/src/rviz/default_plugin/marker_display.cpp
index 79edefd7..dbce8f57 100644
--- a/src/rviz/default_plugin/marker_display.cpp
+++ b/src/rviz/default_plugin/marker_display.cpp
@@ -43,6 +43,10 @@
 
 #include <rviz/default_plugin/marker_display.h>
 
+#if defined(_WIN32) && defined(DELETE)
+  #undef DELETE
+#endif
+
 namespace rviz
 {
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////

diff --git a/src/rviz/ogre_helpers/render_system.cpp b/src/rviz/ogre_helpers/render_system.cpp
index 14d83396..4da6bde9 100644
--- a/src/rviz/ogre_helpers/render_system.cpp
+++ b/src/rviz/ogre_helpers/render_system.cpp
@@ -155,7 +155,11 @@ void RenderSystem::setupDummyWindowId()
 
 void RenderSystem::loadOgrePlugins()
 {
+#ifdef _WIN32
+  std::string plugin_prefix = "";
+#else
   std::string plugin_prefix = get_ogre_plugin_path() + "/";
+#endif
 #ifdef Q_OS_MAC
   plugin_prefix += "lib";
 #endif
